index.html

```diff
@@ -0,0 +1,30 @@
+<!--
+This page is for development purposes. It is the entry point for `webpack-dev-server`.
+It mocks the html environment for `panel/Index.tsx`. Local Storage simulates `globalState`.
+-->
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <link rel="stylesheet" href="index.css">
+        <style>
+            code { font-family: SF Mono, Menlo, Monaco, 'Courier New', monospace; }
+        </style>
+    </head>
+    <body class="vscode-dark">
+        <div id="root" style="margin: 50px 300px; position: relative;"></div><!-- relative for Popover -->
+        <script>
+            acquireVsCodeApi = (() => ({ postMessage: message => {
+                const {command, ...rest} = message;
+                if (command === 'setState') {
+                    const {state} = message;
+                    localStorage.setItem('state', state);
+                    return;
+                }
+                console.log(command, JSON.stringify(rest));
```

Comment:
>console.log [](start = 16, length = 11)

Using console.log can be super annoying to anyone else debugging their extension while this one is installed. The reason is that all console.log calls are displayed while VSCode is being debugged for all extensions (ugh). Our extension has this problem as well and I have a bug to fix it.

A technique that another team did was to add this to "launch.json"
            "name": "Launch Extension",
            ...
            "env": {
                "DEBUGGING_DEVCANVAS": "true"
            }

And they had a small wrapper around console.log which checked to see if they were in debug mode AND this env variable was set:

const gDebug: boolean = (process.execArgv.filter((v) => v.indexOf('--inspect-brk') !== -1).length > 0)
                             && (process.env.DEBUGGING_DEVCANVAS === 'true');

So, the result of this is that they only see their console.log debug spew while debugging THEIR extension. (which is nice)

Reply:
100%. One of my top pet peeves. And I have been the victim of this in almost every team I've been on. :-)

That said, this file is just a tool to help debug things locally. Perhaps the comment on the top isn't obvious enough. This file is outside the `src` folder and thus never goes into the VSIX.

That's a great piece of info about the env flag. Will keep that up my sleeves for the future. :-)

---
In reply to: [441520102](https://github.com/microsoft/sarif-vscode-extension/pull/282#discussion_r441520102) [](ancestors = 441520102)